src/WebJobs.Script.WebHost/Controllers/SwaggerController.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Net;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Microsoft.Azure.WebJobs.Script.WebHost.Filters;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Controllers
+{
+    [AuthorizationLevel(AuthorizationLevel.Function, ScriptConstants.DocumentationResourceName)]
+    public class SwaggerController : ApiController
+    {
+        private readonly ISwaggerDocumentManager _swaggerDocumentManager;
+        private readonly WebScriptHostManager _scriptHostManager;
+
+        public SwaggerController(ISwaggerDocumentManager swaggerDocumentManager, WebScriptHostManager scriptHostManager)
+        {
+            _swaggerDocumentManager = swaggerDocumentManager;
+            _scriptHostManager = scriptHostManager;
+        }
+
+        [HttpGet]
+        [Route("admin/host/swagger/default")]
+        public IHttpActionResult GetGeneratedSwaggerDocument()
+        {
+            var swaggerDocument = _swaggerDocumentManager.GenerateSwaggerDocument(_scriptHostManager.HttpRoutes);
+            return Ok(swaggerDocument);
+        }
+
+        [HttpGet]
+        [Route("admin/host/swagger/current")]
+        public async Task<IHttpActionResult> GetSwaggerDocument()
+        {
+            var swaggerDocument = await _swaggerDocumentManager.GetSwaggerDocumentAsync();
+            if (swaggerDocument == null)
+            {
+                return NotFound();
+            }
+            return Ok(swaggerDocument);
+        }
+
+        [HttpPost]
```

Comment:
There is no partial update to the document. It is more accurately add or overwrite. I feel for that just post is more appropriate